assembl/models/preferences.py

```diff
@@ -775,3 +775,15 @@ def get_preference_data_list(self):
     preference_data_key_list = [p["id"] for p in preference_data_list]
     preference_data_key_set = set(preference_data_key_list)
     preference_data = {p["id"]: p for p in preference_data_list}
+
+
+def includeme(config):
+    """Initialize some preference values"""
+    from ..auth.social_auth import get_active_auth_strategies
+    # Note: This is temporary code, will probably create a new scalar type.
+    settings = config.get_settings()
+    # TODO: Give linguistic names to social auth providers.
+    active_strategies = {
+        k: k for k in get_active_auth_strategies(settings)}
+    active_strategies[''] = _("No special authentication")
+    Preferences.preference_data['authorization_server_backend']['scalar_values'] = active_strategies
```

Comment:
Not exactly. These base values are constructed with ini, then overlaid with whatever is in the Preference sqla objects. But the overlay is sparse, and in known cases the preference_data is absent from the preference objects in assembl. Otherwise, we should actually migrate the preference_data json in Preferences objects in the DB whenever we alter the preferences. I have not done it because I know preference_data is never defined, but that's actually wrong. So this is fragile, but OTH it does allow to change the preferences structure through data (which was given to me as a requirement.)